test/RibbonThetaVaultYearn.js

```diff
@@ -0,0 +1,3541 @@
+const { expect, assert } = require("chai");
+const { BigNumber, constants } = require("ethers");
+const { parseUnits } = require("ethers/lib/utils");
+const { ethers } = require("hardhat");
+const { provider, getContractAt } = ethers;
+const { parseEther } = ethers.utils;
+const { createOrder, signTypedDataOrder } = require("@airswap/utils");
+
+const time = require("./helpers/time");
+const {
+  deployProxy,
+  getDefaultArgs,
+  wmul,
+  wdiv,
+  setupOracle,
+  setOpynOracleExpiryPriceYearn,
+  whitelistProduct,
+  parseLog,
+  mintToken,
+  setAssetPricer,
+  getAssetPricer,
+} = require("./helpers/utils");
+const moment = require("moment-timezone");
+moment.tz.setDefault("UTC");
+
+let owner, user, manager, counterparty, feeRecipient;
+let adminSigner,
+  userSigner,
+  ownerSigner,
+  managerSigner,
+  counterpartySigner,
+  feeRecipientSigner;
+
+const WETH_ADDRESS = "0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2";
+const Y_WETH_ADDRESS = "0xa9fE4601811213c340e850ea305481afF02f5b28";
+const WBTC_ADDRESS = "0x2260FAC5E5542a773Aa44fBCfeDf7C193bc2C599";
+const USDC_ADDRESS = "0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48";
+//const WBTC_OWNER_ADDRESS = "0xCA06411bd7a7296d7dbdd0050DFc846E95fEBEB7";
+//const USDC_OWNER_ADDRESS = "0x47ac0fb4f2d84898e4d9e7b4dab3c24507a6d503";
+
+const YEARN_WETH_PRICER = "0x7b7a7fA8e7A86F0100E72E815E65006052364F48";
+const CHAINLINK_WETH_PRICER = "0xAC05f5147566Cc949b73F0A776944E7011FabC50";
+
+const OTOKEN_FACTORY = "0x7C06792Af1632E77cb27a558Dc0885338F4Bdf8E";
+const MARGIN_POOL = "0x5934807cC0654d46755eBd2848840b616256C6Ef";
+const SWAP_ADDRESS = "0x4572f2554421Bd64Bef1c22c8a81840E8D496BeA";
+const YEARN_REGISTRY_ADDRESS = "0x50c1a2eA0a861A967D9d0FFE2AE4012c2E053804";
+const SWAP_CONTRACT = "0x4572f2554421Bd64Bef1c22c8a81840E8D496BeA";
+const TRADER_AFFILIATE = "0xFf98F0052BdA391F8FaD266685609ffb192Bef25";
+
+const OPTION_DELAY = 60 * 60; // 1 hour
+const LOCKED_RATIO = parseEther("0.9");
+const WITHDRAWAL_BUFFER = parseEther("1").sub(LOCKED_RATIO);
+const gasPrice = parseUnits("1", "gwei");
+
+const PUT_OPTION_TYPE = 1;
+const CALL_OPTION_TYPE = 2;
+
+describe("RibbonThetaVaultYearn", () => {
+  behavesLikeRibbonOptionsVault({
+    name: `Ribbon ETH Yearn Theta Vault (Call)`,
+    tokenName: "Ribbon ETH Yearn Theta Vault",
+    tokenSymbol: "rETH-THETA-YEARN",
+    asset: WETH_ADDRESS,
+    assetContractName: "IWETH",
+    collateralContractName: "IYearnVault",
+    strikeAsset: USDC_ADDRESS,
+    collateralAsset: Y_WETH_ADDRESS,
+    depositAsset: WETH_ADDRESS,
+    wrongUnderlyingAsset: WBTC_ADDRESS,
+    wrongStrikeAsset: WBTC_ADDRESS,
+    firstOptionStrike: 63000,
+    secondOptionStrike: 64000,
```

Comment:
We should use strike prices that look like ETH's prices

Reply:
yeah, was just going with convention in ribbonthetavault.js

https://github.com/ribbon-finance/structured-products/blob/3fb63f11c7e4d0ef064f963c74bcfd78f4838809/test/RibbonThetaVault.js#L65